### PR TITLE
HG-736 Add in support for forwarding headers

### PR DESCRIPTION
**Pull request by kenkouot**
This PR is part of a fix for HG-736. At a high level, there is a race between our purges in Celery, the MW edit API and refreshing/viewing an article in Mercury. In order to prevent a user who has just edited from getting stale content (and also for logged in users), we're going to use the same mechanism MW does and simply ignore cache for requests in which the `wikicities_session` cookie is present.

Previously, calls to the internal Varnish were not setting any cookies. This sets the user session cookie if it is available.

### CODE REVIEW

**Comment by lizlux** [comment]
`server/facets/api/article.ts(34,11): 2304 Cannot find name 'ArticleRequestParams'.
server/facets/showArticle.ts(21,11): 2304 Cannot find name 'ArticleRequestParams'.`

File: server/facets/api/article.ts [hunk 1]
```diff
@@ -30,21 +30,31 @@ function isRequestForRandomTitle (query: any): boolean {
...
-	var wikiDomain = Utils.getCachedWikiDomainName(localSettings, request.headers.host);
+	var wikiDomain = Utils.getCachedWikiDomainName(localSettings, request.headers.host),
+		params: ArticleRequestParams = {
+			wikiDomain: wikiDomain,
+			title: request.params.articleTitle,
+			redirect: request.params.redirect
+		},
+		article: Article.ArticleRequestHelper;
+
+		if (request.state.wikicities_session) {
```
Comment (lizlux) [inline server/facets/api/article.ts]: I wish this code was more self-documenting, but I understand why it isn't - b/c it is taking this pattern from MW. Let's add a comment here so we know why we're adding this cookie header to the request.

**Comment by lizlux** [comment]
This would probably be a good time to add some unit tests.

File: server/facets/api/article.ts (hunk 1 above)
Comment (rogatty) [inline server/facets/api/article.ts]: Indentation is off here.

**Comment by rogatty** [comment]
@lizlux: I don't get [these warnings](https://github.com/Wikia/mercury/pull/902#issuecomment-110528105). Do you, still?

`ArticleRequestParams` is defined in `mercury-server.d.ts`.
Both `server/facets/api/article.ts` and `server/facets/showArticle.ts` refer to `lib/Article.ts` which in turn refers to `mercury-server.d.ts`.

Maybe we should add this last reference to both `article.ts` and `showArticle.ts`? Would be nice to standarize approach and add it to the guidelines. I work on this code for some time now and I'm still not sure when do I have to add references :see_no_evil:

**Comment by lizlux** [comment]
@rogatty, not seeing them in dev, so you're good. 

I'd be all for adding this to the guidelines.

**Pull request merged**